earthfile2llb/converter.go

```diff
@@ -2416,6 +2429,65 @@ func (c *Converter) internalRun(ctx context.Context, opts ConvertRunOpts) (pllb.
 	}
 }
 
+// detectAWSCredentials determines whether AWS envs are set and/or ~/.aws
+// exists. These are then used when fetching the secrets during a build.
+func (c *Converter) detectAWSCredentials(ctx context.Context) ([]llb.RunOption, []string, error) {
```

Comment:
I think the logic related to fetching credentials could be moved somewhere under `util/...`, in order to limit the converter code to llb-specific functionality.